mix/ContractCallDataEncoder.cpp

```diff
@@ -55,54 +63,92 @@ void ContractCallDataEncoder::encode(QFunctionDefinition const* _function)
 	m_encodedData.insert(m_encodedData.end(), hash.begin(), hash.end());
 }
 
+void ContractCallDataEncoder::encodeArray(QJsonArray const& _array, QList<int> _dim, SolidityType const& _type, bytes& _content)
+{
+	size_t offsetStart = _content.size();
+	if (_dim[0] == -1)
+	{
+		bytes count = bytes(32);
+		toBigEndian((u256)_array.size(), count);
+		_content += count; //reserved space for count
+	}
+	_dim.pop_front();
+
+	int k = 0;
+	for (QJsonValue const& c: _array)
+	{
+		if (c.isArray())
+		{
+			if (_dim[0] == -1)
+				m_dynamicOffsetMap.push_back(std::make_pair(m_dynamicData.size() + offsetStart + 32 + k * 32, m_dynamicData.size() + _content.size()));
+			encodeArray(c.toArray(), _dim, _type, _content);
+		}
+		else
+		{
+			// encode single item
+			if (c.isDouble())
+				encodeSingleItem(QString::number(c.toDouble()), _type, _content);
+			else if (c.isString())
+				encodeSingleItem(c.toString(), _type, _content);
+		}
+		k++;
+	}
+}
+
 void ContractCallDataEncoder::encode(QVariant const& _data, SolidityType const& _type)
 {
-	u256 count = 1;
 	QStringList strList;
+	vector<int> dim;
```

Comment:
No, you pass `QList<int> dim` to encodeArray() because `vector<int> dim` has been shadowed.